src/ert/_c_wrappers/enkf/enkf_fs.py

```diff
@@ -244,6 +260,25 @@ def load_gen_data(
             raise KeyError(f"Unable to load GEN_DATA for key: {key}")
         return np.stack(result).T, loaded
 
+    def load_gen_data_as_df(
+        self, keys: List[str], realizations: List[int]
+    ) -> DataFrame:
+        dfs = []
+        for key in keys:
+            data, realizations = self.load_gen_data(key, realizations)
+            x_axis = [*range(data.shape[0])]
+            multi_index = MultiIndex.from_product(
+                [[key], x_axis], names=["data_key", "axis"]
+            )
+            dfs.append(
+                DataFrame(
+                    data=data.reshape(len(x_axis), len(realizations)),
```

Comment:
Same thing here, do we need to reshape?

Reply:
output of enkf_fs.load_gen_data is a 3D numpy array, but pandas expect a 2D array

Reply:
You were correct, the 3D array was old behaviour. Reshape removed :+1: